src/intersection-observer.js

```diff
@@ -81,20 +81,29 @@ export function getIntersectionChangeEntry(element, owner, viewport) {
       'Negative dimensions in element.');
   // Building an IntersectionObserverEntry.
 
-  let intersectionRect = element;
+  // The element is relative to (0, 0), while the viewport moves. So, we must
+  // adjust.
+  const boundingClientRect = moveLayoutRect(element, -viewport.left,
```

Comment:
We [need](https://github.com/ampproject/amphtml/issues/5090) this PR now (I screwed up `IntersectionObserver` by not translating the coords). The fixed position stuff is [blocking](https://github.com/ampproject/amphtml/issues/3812) `<amp-fx-flying-carpet>`.